pages/knowledge/web-components/README.md

```diff
@@ -0,0 +1,49 @@
+>
+> <cite>[Using custom elements (mdn)](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements)</cite>
+
+### Shadow DOM
+
+> An important aspect of custom elements is encapsulation, because a custom element, by definition, is a piece of reusable functionality: it might be dropped into any web page and be expected to work. So it's important that code running in the page should not be able to accidentally break a custom element by modifying its internal implementation. Shadow DOM enables you to attach a DOM tree to an element, and have the internals of this tree hidden from JavaScript and CSS running in the page.
+>
+> <cite>[Using shadow DOM (mdn)](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM)</cite>
+
+### HTML templates
+
+> When you have to reuse the same markup structures repeatedly on a web page, it makes sense to use some kind of a template rather than repeating the same structure over and over again. This was possible before, but it is made a lot easier by the HTML \<template\> element. This element and its contents are not rendered in the DOM, but it can still be referenced using JavaScript.
+>
+> <cite>[Using templates and slots
+> (mdn)](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_templates_and_slots)</cite>
+
+## Browser support
+
+Overall, above technologies are well supported in current browsers. Though aspects like _declarative shadow DOM_ have [just recently landed](https://caniuse.com/declarative-shadow-dom) in all major browsers at the time of writing (April 2024). Declarative shadow DOM allows us to render shadow DOM server-side rather than relying on client-side JavaScript.
+
+## Web Components and accessibility
+
+Web components have great potential to help us improve accessibility as they are framework-agnostic. This allows us to _share_ accessible components with everyone, independent of which UI framework they use.
+
+However, when using shadow DOM, we need to be aware of the limitations of crossing "shadow boundaries". Specifically, we cannot yet easily link elements outside of a shadow root with elements within. There is ongoing specification work regarding associating [labels and form elements in particular](https://github.com/whatwg/html/issues/3219) or [any DOM elements using ARIA attributes](https://github.com/WICG/aom/issues/192), to give two examples.
+
+Manuel Matuzović gives the following recommendation in his [great list of FAQ](https://www.matuzo.at/blog/2023/web-components-accessibility-faq):
+
+> If all your relationships for an element happen exclusively in light DOM or shadow DOM and you don't try to cross boundaries, working with ARIA is not a problem.
+
+Additionally, some testing tools are not yet able to handle shadow DOM. In most cases, this should be [straight-forward to fix](https://github.com/hinderlingvolkart/h123/pull/5).
```

Comment:
```suggestion
Additionally, some testing tools are not yet able to handle shadow DOM. In most cases, this should be straight-forward to fix.
```
I think this link is a bit too specific and therefore kind of abstract for most people. Unless there is a link to a more general page with instructions or tips, I wouldn't include a link for now.

Reply:
@christophdubach, this is a very good point. WDYT of this more specific example? https://github.com/Access4all/adg/pull/439/commits/64aa3970ac0c97cd089b85b8fff2573ee206dc44

Reply:
Simple and easy to understand  👍🏼